web/src/settings_components.ts

```diff
@@ -1556,25 +1556,25 @@ export function set_group_setting_widget_value(
     pill_widget: GroupSettingPillContainer,
     property_value: GroupSettingValue,
 ): void {
-    pill_widget.clear();
+    pill_widget.clear(true);
```

Comment:
We use `set_group_setting_widget_value` function for two cases -
- Setting the pills when initializing the widgets.
- Discarding the changes. Live updating the widgets also uses same code to discard the changes.

So, it is fine to just not check whether to show the save-discard when adding and removing pills here.